src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -254,16 +394,78 @@ unique_ptr<CatalogEntry> TableCatalogEntry::AddColumn(ClientContext &context, Ad
 	                                      new_storage);
 }
 
-unique_ptr<CatalogEntry> TableCatalogEntry::RemoveColumn(ClientContext &context, RemoveColumnInfo &info) {
+void DependsOnColumn(const ParsedExpression &expr, const string &column, bool &result) {
+	if (expr.type == ExpressionType::COLUMN_REF) {
+		auto column_ref = (ColumnRefExpression &)expr;
+		auto &column_name = column_ref.GetColumnName();
+		if (column_name == column) {
+			result = true;
+		}
+	}
+	ParsedExpressionIterator::EnumerateChildren(
+	    expr, [&](const ParsedExpression &child) { DependsOnColumn(child, column, result); });
+}
+
+unique_ptr<CatalogEntry> TableCatalogEntry::RemoveGeneratedColumn(ClientContext &context, RemoveColumnInfo &info,
+                                                                  idx_t index) {
 	auto create_info = make_unique<CreateTableInfo>(schema->name, name);
-	create_info->temporary = temporary;
-	idx_t removed_index = GetColumnIndex(info.removed_column, info.if_exists);
+
+	// Copy all the columns
+	for (idx_t i = 0; i < columns.size(); i++) {
```

Comment:
Like with the remove this should be mostly the same code as the regular `RemoveColumn`, no? Can we unify this into a single function?